src/services/client_factory.py

```
@@ -0,0 +1,123 @@
+except ImportError:
+    # Fallback for when module is imported directly (not as package)
+    from config import get_logger
+
+# Configure logger for this module
+logger = get_logger(__name__)
+
+
+class SupabaseClientProtocol(Protocol):
+    """Protocol for Supabase client interface."""
+
+    def table(self, table_name: str): ...
```

Pylint (3.3.7)

[convention] Missing function or method docstring

(C0116)